test/test_grad.jl

```diff
@@ -24,6 +32,12 @@ update_chainrules_primitives!()
 # end
 
 
+@testset "grad: no_rrule" begin
+    @test is_chainrules_primitive(Tuple{typeof(chain_foo), Float64}) == true
+    @test is_chainrules_primitive(Tuple{typeof(chain_foo), Real}) == false
+    @test is_chainrules_primitive(Tuple{typeof(chain_foo), Float64}) == true
```

Comment:
Did you mean to test some number type here that was not a subtype of `Real` ?
Like:
```suggestion
    @test is_chainrules_primitive(Tuple{typeof(chain_foo), ComplexF64}) == true
```
Which should hit `Number`

Or maybe subtype of `Real` that was not `Float64`

```suggestion
    @test is_chainrules_primitive(Tuple{typeof(chain_foo), Float32}) == false
```
Which should be blocked by opt out `Real`

Reply:
My idea in this test is that `no_rrule` blocks only the `rrule` for that exact type, not its subtypes or supertypes, so yeah, it's the same thing but expressed differently. Given that signature usually consists of concrete types only, I actually like your example more.